sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/plans/logical/statsEstimation/EstimationUtils.scala

```diff
@@ -114,4 +115,183 @@ object EstimationUtils {
     }
   }
 
+  /**
+   * Returns overlapped ranges between two histograms, in the given value range [newMin, newMax].
+   */
+  def getOverlappedRanges(
+      leftHistogram: Histogram,
+      rightHistogram: Histogram,
+      newMin: Double,
+      newMax: Double): Seq[OverlappedRange] = {
```

Comment:
how about `upperBound`/`lowerBound`? It's hard to understand the meaning of `new` by looking at this method.

Reply:
`max`/`min` is also fine

Reply:
yea I think `upperBound/lowerBound` is better.